### ISSUE TITLE
False Positive for strict-type-predicates Rule

### ISSUE THREAD

**Issue by chrismbarr** [palantir/tslint#3168]
### Bug Report

- __TSLint version__: 5.6.0
- __TypeScript version__: 2.4.2
- __Running TSLint via__: gulp-tslint

#### TypeScript code being linted

```ts
public static isNullOrEmpty = <TType>(list: TType[] | undefined | null): boolean =>
    list == undefined || list === null;
//                       ~~~~~~~~~~~~~ [strict-type-predicates  Expression is always false]
```

with `tslint.json` configuration:

```json
"strict-type-predicates": true,
```

#### Actual behavior
The `strict-type-predicates` rule reports the error "Expression is always false" when the type of `list` could be a generic type, `undefined` or `null` when I do a `list === null` check.

#### Expected behavior
Since this could also be other types other than `null` saying that this "expression is always false" seems wrong.

Am I missing something here?

**Comment by ajafff** [comment]
> Am I missing something here?

Indeed, you are: `list == undefined` also removes `null` from the union

**Comment by chrismbarr** [comment]
oh right, good catch!  Thanks.
